### ISSUE TITLE
`significant_drop_tightening` suggests incomplete binding

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#15602]
### Summary

When the temporary with significant `Drop` is only used once, and that use is a let statement, the lints suggests inlining the the former into the latter, but the suggestion doesn't forgets all the "modifiers" of the binding, most notably the `mut`.

That's because, to creat the suggestion, the lint doesn't actually inline the former into the latter, but instead reconstructs the latter _around_ the former, and in that process only handles the ident of the binding.

One possible approach would be to make the reconstruction more robust, but an imo better solution is to actually do the inlining, to avoid the complexity altogether.

### Reproducer

I tried this code:

```rust
fn main() {
    use std::io::{Read, Take, stdin};

    let stdin = std::io::stdin().lock();
    let mut shadowing @ Take { .. } = stdin.take(40);
    do_heavy_computation_that_takes_time(())
}

fn do_heavy_computation_that_takes_time() {}
```

I expected to see this happen:
```
error: temporary with significant `Drop` can be early dropped
  --> tests/ui/significant_drop_tightening.rs:144:9
   |
LL |   fn merge_into_complex_bindings() {
   |  __________________________________-
LL | |     use std::io::{Read, Take, stdin};
LL | |
LL | |     let stdin = std::io::stdin().lock();
   | |         ^^^^^
...  |
LL | |     do_heavy_computation_that_takes_time(())
LL | | }
   | |_- temporary `stdin` is currently being dropped at the end of its contained scope
   |
   = note: this might lead to unnecessary resource contention
help: merge the temporary construction with its single usage
   |
LL ~     
LL +     let mut shadowing @ Take { .. } = std::io::stdin().lock().take(40);
LL |
LL ~     
   |
```

Instead, this happened:
```
<same lint message and note>
help: merge the temporary construction with its single usage
   |
LL ~     
LL +     let shadowing = std::io::stdin().lock().take(40);
LL |
LL ~     
   |
```

### Version

```text
(playground, stable 1.89.0)
```

### Additional Labels

@rustbot label I-suggestion-causes-error
@rustbot claim
